addons/hr/models/hr.py

```diff
@@ -334,6 +334,27 @@ def get_import_templates(self):
             'template': '/hr/static/xls/hr_employee.xls'
         }]
 
+    def _post_author(self):
+        """
+        When a user updates his own employee's data, all operations are performed
+        by super user. However, tracking messages should not be posted as OdooBot
+        but as the actual user.
+        This method is used in the overrides of `_message_log` and `message_post`
+        to post messages as the correct user.
+        """
+        real_user = self.env.context.get('binary_field_real_user')
```

Comment:
What's that ????

Doesn't seem to be used though.

Reply:
If it's not useful now (old code), please remove this.

Reply:
As far as I see, this context key has been added by https://github.com/odoo/odoo/pull/26635/files only a few months ago. And it does exactly what I need (keep the real user when the record is "sudoed" from `SELF_WRITABLE_FIELDS` (this [line](https://github.com/Tecnativa/odoo/blob/8fc52a4bda011965239d2f28c37b8c80005e8782/odoo/addons/base/models/res_users.py#L452) to post tracking messages as the real user).
But I agree it's weird and confusing to see that here, more specifically: why is it called `binary_field_[...]` ?

Reply:
Yep ok I get the point ;)

Reply:
But due to the butterfly effect, you've just created a new separatist group in the world.

Congratulations.

Reply:
I don't like that either. How would you improve this ?